EventsExpress.Core/Services/LocationService.cs

```diff
@@ -43,21 +44,33 @@ public async Task<Guid> Create(LocationDto locationDTO)
 
         public LocationDto LocationByPoint(Point point)
         {
-            var locationDTO = Mapper.Map<LocationDto>(Context.EventLocations
+            var locationDto = Mapper.Map<LocationDto>(Context.Locations
                 .Where(e =>
                     e.Point.X == point.X &&
                     e.Point.Y == point.Y)
                 .FirstOrDefault());
-            return locationDTO;
+            return locationDto;
         }
 
         public LocationDto LocationByURI(string uri)
         {
-            var locationDTO = Mapper.Map<LocationDto>(Context.EventLocations
+            var locationDTO = Mapper.Map<LocationDto>(Context.Locations
                 .Where(e =>
                    e.OnlineMeeting == uri)
                 .FirstOrDefault());
             return locationDTO;
         }
+
+        public async Task<Guid> AddLocationToUser(LocationDto locationDto)
+        {
+            if (locationDto.Type == LocationType.Map)
+            {
+                return LocationByPoint(locationDto.Point)?.Id ?? await Create(new LocationDto { Id = locationDto.Id, Point = locationDto.Point, OnlineMeeting = null, Type = locationDto.Type });
+            }
+            else
+            {
+                return LocationByURI(locationDto.OnlineMeeting)?.Id ?? await Create(new LocationDto { Id = locationDto.Id, Point = null, OnlineMeeting = locationDto.OnlineMeeting, Type = locationDto.Type });
+            }
```

Comment:
this does not make any sense for the user's location. I think we should throw here